src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -1220,16 +1222,22 @@ function applyLoadoutMods(
       const pluggingSteps = createPluggingStrategy(defs, item, assignments);
       const assignmentSequence = pluggingSteps.filter((assignment) => assignment.required);
       infoLog(TAG, 'Applying mods', assignmentSequence, 'to', item.name);
-      if (assignmentSequence) {
```

Comment:
?? this always returns an array. so. weird check.

Reply:
Hm I actually thought/hoped we had a lint for this sort of thing.

Reply:
Ah now I remember. It's `@typescript-eslint/strict-boolean-expressions` / `@typescript-eslint/no-unnecessary-condition` which I could never quite tune to be permissive enough with our existing code.

Reply:
(and to make that work we need `noUncheckedIndexedAccess` which is its own problem)

Reply:
It's also entirely possible that I was at some point convinced that empty array was falsy, given how many of these checks there are.

Reply:
(it doesn't help that the definitions say a lot of things are non-optional when they can definitely be missing)